src/screenComponents/radarView.cpp

```diff
@@ -650,6 +650,14 @@ void GuiRadarView::drawObjects(sp::RenderTarget& renderer)
         break;
     }
 
+    std::vector objects_to_draw(std::begin(visible_objects), std::end(visible_objects));
+    std::sort(std::begin(objects_to_draw), std::end(objects_to_draw), [](const auto& lhs, const auto& rhs)
+    {
+        const auto lhsLayer = lhs->getRadarLayer();
+        const auto rhsLayer = rhs->getRadarLayer();
+        return lhsLayer < rhsLayer || (lhsLayer == rhsLayer && lhs->canHideInNebula() && !rhs->canHideInNebula());
```

Comment:
First order by layers, and if in the same layers, put objects that can hide in nebulas "lower".